app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any
+from dataclasses import dataclass
+
+
 class Dictionary:
-    pass
+    _DEFAULT_LENGTH = 8
+    _LENGTH_RESIZE_COEFFICIENT = 2 / 3
+    _RESIZE_MULTIPLY_COEFFICIENT = 2
+
+    @dataclass()
+    class _Node:
+        key: Any
+        value: object
+        hash_: int
+
+    def __init__(self):
+        self._container = [None for _ in range(self._DEFAULT_LENGTH)]
+        self._length = 0
+        self._current_length = len(self._container)
+
+    def __setitem__(self, key, value):
+        capacity = int(self._LENGTH_RESIZE_COEFFICIENT * self._current_length)
+        hash_ = hash(key)
+        index = hash_ % self._current_length
+
+        while self._container[index % self._current_length] is not None:
+            if index == self._current_length:
+                index = 0
+
+            if self._container[index].key == key and hash_ == hash(key):
+                self._container[index].value = value
+                return
+            index += 1
+
+        if capacity <= self._length:
+            self.resize()
+
+        self._container[index % self._current_length] =\
+            self._Node(key, value, hash_)
+        self._length += 1
+
+    def __getitem__(self, key):
```

Comment:
It is not right, because first you need to address to index of `hash(key)`, you loop over whole container instead.